elasticsearchprobe/elasticsearchprobe.go

```diff
@@ -0,0 +1,49 @@
+package elasticsearchprobe
+
+import (
+	"context"
+	"crypto/tls"
+	"net/http"
+
+	"github.com/opensearch-project/opensearch-go"
+	"github.com/pkg/errors"
+)
+
+type ElasticsearchProbe struct {
+	name string
+	url  string
+}
+
+// NewElasticsearchProbe instantiate a new elasticsearch probe:
+// - name: probe name
+// - url : connection string with the form "http://username:password@example.com"
+func NewElasticsearchProbe(name, url string) ElasticsearchProbe {
+	return ElasticsearchProbe{
+		name: name,
+		url:  url,
+	}
+}
+
+func (p ElasticsearchProbe) Name() string {
+	return p.name
+}
+
+func (p ElasticsearchProbe) Check(_ context.Context) error {
+	cfg := opensearch.Config{
+		Addresses: []string{p.url},
+		Transport: &http.Transport{
```

Comment:
Could you also protect this probe from redirect attack (like we did for the Sendinblue probe ?)

Reply:
> Could you also protect this probe from redirect attack (like we did for the Sendinblue probe ?)

There is no CheckRedirect option in opensearch.Config. According to [this answer](https://github.com/elastic/go-elasticsearch/issues/149#issuecomment-624488457) it might no be something possible to change with this client